lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,13 +153,87 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(connection, logger, connection_options, config)
+          super
+          if @connection
+            @verified = true
+            @raw_connection = @connection
+          end
+          # Ensure that we're treating prepared_statements in the same way that Rails 7.1 does
+          @prepared_statements = self.class.type_cast_config_to_boolean(
+            @config.fetch(:prepared_statements) { default_prepared_statements }
+          )
+        end
+
+        def connect!
```

Comment:
Q: what is the reason for overriding this method ? (is it specifically because of `@verified` flag ?